### PR TITLE
Create InsertSongbook API

### PR DESCRIPTION
**Pull request by brandonxia01**
Setting up some code structure for APIs. Layers look kind of like this

API Layer (index.ts)
- validates the incoming request
- calls "service" layer for all the logic
- returns response back to client

Service Layer (services/SongsService.ts)
- handles business logic
- calls database layer
- returns data back to API layer

Database Layer (db/SongsDb.ts)
- takes in parameters for queries
- builds SQL queries and makes the call to the database
- returns the rows mapped into the JSON data objects

Helpers 
- Helper functions to extract some extra code

Next PR - to try to create some test cases

### CODE REVIEW

File: src/db/DbQueries.ts [hunk 1]
```diff
@@ -5,17 +5,28 @@
...
+export function buildInsertSongbookQuery(
+  id: string,
+  fullName: string,
+  staticMetadataLink: string,
+  imageUrl: string
+): string {
+  return `
+        INSERT INTO songbooks 
+        (id, full_name, static_metadata_link, image_url, inserted_dt, updated_dt)
+        VALUES ('${id}', '${fullName}', '${staticMetadataLink}','${imageUrl}', now(), now())
```
[the remark this answers is not in the document]
Comment (brandonxia01) [reply src/db/DbQueries.ts:17]: It looks like this `2023-01-30 12:44:24.308 -0500` -5 is EST which could be bc our server is in NYC or bc my computer is in Cleveland (both are EST). It doesn't matter much tho, you can translate whatever timezone back to utc if we needed

File: src/index.ts [hunk 2]
```diff
@@ -24,10 +29,44 @@ app.get('/db', async (req, res) => {
...
+    validateInsertSongbookRequest(dbSongbook);
+    insertSongbookMethod(dbSongbook).then((val) => {
+      console.log(`Returning ${val}`);
+      res.status(200).send(val ?? {});
+    });
+  } catch (e: any) {
+    handleErrorsAndReturn(e, res);
+  }
+});
+
```
[the remark this answers is not in the document]
Comment (brandonxia01) [reply src/index.ts]: Ah gotcha, cool - updated

File: src/db/DbQueries.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiTo0o) [reply src/db/DbQueries.ts:17]: i see, sounds good :D

**Review by MiTo0o — APPROVED** [review]
lgtm

**Pull request merged**